### PR TITLE
WMS/WFS/WCS - separate core and gui of data items

### PR DESCRIPTION
**Pull request by wonder-sk**
First PR to follow up #30234 - this one removes deprecated QgsDataItem::actions() implementations from wms, wfs and wcs providers.

There are no actual functionality changes - just refactoring.

### CODE REVIEW

File: src/providers/wcs/qgswcsdataitemguiprovider.cpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+#include "qgsnewhttpconnection.h"
+#include "qgsowsconnection.h"
+
+
+void QgsWcsDataItemGuiProvider::populateContextMenu( QgsDataItem *item, QMenu *menu, const QList<QgsDataItem *> &, QgsDataItemGuiContext )
+{
+  if ( QgsWCSRootItem *rootItem = qobject_cast< QgsWCSRootItem * >( item ) )
+  {
+    QAction *actionNew = new QAction( tr( "New Connection…" ), this );
+    setItemForAction( actionNew, rootItem );
```
Comment (nyalldawson) [inline src/providers/wcs/qgswcsdataitemguiprovider.cpp]: Couldn't you use lambda capturing rootItem here instead? Seems cleaner to me

File: src/providers/wcs/qgswcsdataitemguiprovider.cpp [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  }
+
+  if ( QgsWCSConnectionItem *connItem = qobject_cast< QgsWCSConnectionItem * >( item ) )
+  {
+    QAction *actionEdit = new QAction( tr( "Edit…" ), this );
+    setItemForAction( actionEdit, connItem );
+    connect( actionEdit, &QAction::triggered, this, &QgsWcsDataItemGuiProvider::editConnection );
+    menu->addAction( actionEdit );
+
+    QAction *actionDelete = new QAction( tr( "Delete" ), this );
```
Comment (nyalldawson) [inline src/providers/wcs/qgswcsdataitemguiprovider.cpp]: Not related to this pr, but delete really should have a confirmation prompt

**Review by nyalldawson — COMMENTED** [review]
Looks good!

**Comment by wonder-sk** [comment]
Thanks @nyalldawson for the review - I have addressed your comments

**Pull request merged**